core/chain_manager.go

```diff
@@ -92,6 +92,7 @@ type ChainManager struct {
 	futureBlocks *BlockCache
 
 	quit chan struct{}
+	wg   sync.WaitGroup
```

Comment:
Sure thing, will cherry pick the commits from @fjl branch then to leave this out for a separate PR.